doc/source/whatsnew/v1.3.0.rst

```diff
@@ -153,6 +153,57 @@ Other API changes
 
 Deprecations
 ~~~~~~~~~~~~
+
+:meth:`~DataFrame.groupby` no longer ignores ``group_keys`` for transform-like ``apply``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:meth:`~DataFrame.groupby` will no longer ignore the ``group_keys`` argument for functions passed to ``apply`` that return like-indexed outputs (:issue:`34809`, :issue:`31612`, :issue:`14927`, :issue:`13056`).
+Previous versions of pandas would add the group keys only when the result from the applied function had a different index to the input.
+
+*Previous behavior*:
+
+.. code-block:: python
+
+   >>> # pandas 1.2
+   >>> df = pd.DataFrame({"A": [1, 2, 2], "B": [1, 2, 3]})
+   >>> df
+      A  B
+   0  1  1
+   1  2  2
+   2  2  3
+   >>> df.groupby("A").apply(lambda x: x.rename(np.exp))  # Different index
+               A  B
+   A
+   1 1.000000  1  1
+   2 2.718282  2  2
+     7.389056  2  3
+
+   >>> df.groupby("A").apply(lambda x: x)  # Same index
+      A  B
+   0  1  1
+   1  2  2
+   2  2  3
+
+In this future this behavior will change to always respect ``group_keys``, which defaults to True.
```

Comment:
ok yeah put some of this in the note in the user docs (or you can just put a link in the user docs back to here is ok too)